homeassistant/components/vizio/config_flow.py

```diff
@@ -0,0 +1,131 @@
+"""Config flow for Vizio."""
+
+import logging
+from typing import Any, Dict
+
+from pyvizio import VizioAsync
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import (
+    CONF_ACCESS_TOKEN,
+    CONF_DEVICE_CLASS,
+    CONF_HOST,
+    CONF_NAME,
+)
+from homeassistant.data_entry_flow import AbortFlow
+
+from . import validate_auth
+from .const import (
+    CONF_VOLUME_STEP,
+    DEFAULT_DEVICE_CLASS,
+    DEFAULT_NAME,
+    DEFAULT_VOLUME_STEP,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def update_schema_defaults(input_dict: Dict[str, Any]) -> vol.Schema:
+    """Update schema defaults based on user input/config dict. Retains info already provided for future form views."""
+    return vol.Schema(
+        {
+            vol.Required(
+                CONF_NAME, default=input_dict.get(CONF_NAME, DEFAULT_NAME)
+            ): str,
+            vol.Required(CONF_HOST, default=input_dict.get(CONF_HOST)): str,
+            vol.Optional(
+                CONF_DEVICE_CLASS,
+                default=input_dict.get(CONF_DEVICE_CLASS, DEFAULT_DEVICE_CLASS),
+            ): vol.All(str, vol.Lower, vol.In(["tv", "soundbar"])),
+            vol.Optional(
+                CONF_ACCESS_TOKEN, default=input_dict.get(CONF_ACCESS_TOKEN, "")
+            ): str,
+            vol.Optional(
+                CONF_VOLUME_STEP,
+                default=input_dict.get(CONF_VOLUME_STEP, DEFAULT_VOLUME_STEP),
+            ): vol.All(vol.Coerce(int), vol.Range(min=1, max=10)),
+        }
+    )
+
+
+class VizioConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Vizio config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self) -> None:
+        """Initialize config flow."""
+        self.import_schema = None
+        self.user_schema = None
+
+    async def async_step_user(
+        self, user_input: Dict[str, Any] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initialized by the user."""
+        errors = {}
+
+        if user_input is not None:
+            await self.async_set_unique_id(unique_id=user_input[CONF_HOST])
```

Comment:
We should use the serial for unique_id since that's the most future proof.

Reply:
Removing it since I can't get the serial until after the device is initialized and can be connected to

Reply:
Remove this too.